src/main/java/com/teragrep/rlp_03/context/ConnectionContextImpl.java

```diff
@@ -141,6 +141,7 @@ public void handleEvent(SelectionKey selectionKey) {
                 }
                 catch (RejectedExecutionException ree) {
                     LOGGER.error("executorService.submit threw <{}> for read", ree.getMessage());
+                    throw new RuntimeException("Rejected execution on executorService.submit: " + ree);
```

Comment:
Keep rethrown type as the same so
throw ree;

Reply:
fix: c41251fc4c46651b03a006a27ae5385c742e04c5